### ISSUE TITLE
RTX 2080 ti has low performance

### ISSUE THREAD

**Issue by blaine141** [AlexeyAB/darknet#4202]
I have been training on a Titan X Pascal and each iteration takes about 7 seconds. I then started using another machine with a 2080 ti. I saw lower performance on the 2080 ti machine until the tensor cores kicked in, then there was equal performance. Both machines have an ssd. The Titanx machine has 64 cores with 2 Intel(R) Xeon(R) CPU E5-2697A v4 @ 2.60GHz with 4 Titan Xs although I am only using one with GPU and CUDNN set to true and the 2080 ti machine has 16 cores with one Intel(R) Core(TM) i7-7820X CPU @ 3.60GHz and 2 2080 TIs although I am only using one with GPU, CUDNN, and CUDNN_HALF set to true. Below is my configuration file.
[yolov3.cfg](https://github.com/AlexeyAB/darknet/files/3796378/yolov3-boards.txt) 

What am I doing wrong? Is the 2080 TI slower than the Titan X Pascal and if not does the cpu difference explain it? Seems like there should be a large difference in performance

**Comment by AlexeyAB** [comment]
* Titan X - 10 TFlops-sp
* 2080 TI  - 12 TFlops-sp and 94 TFlops-TensorCores - it should be faster.




What load-time do you get for several iterations? Is it always 0 sec?



![image](https://user-images.githubusercontent.com/4096485/68020299-1d25be80-fcaf-11e9-98bf-91679a792fac.png)

**Comment by blaine141** [comment]
That number varies but is around 0.25 on both machines. At first I had load times around 25 seconds on the 2080 machine but that number went back to normal once I disabled Open cv. I probably have the wrong version of opencv. The loading duration obviously jumps up when resizing. The difference in time was noticed while processing a batch. The titanx was running through subdivions slightly faster than the other and had slightly lower batch times. One other thing to note is the 2080 showed about 30% gpu utilization.

**Comment by AlexeyAB** [comment]
Do you use the same batch and subdivisions on both PC?

**Comment by blaine141** [comment]
Yes. The config and training data is a exact copy between the machines.
Should also note cuda 10.1 and driver 418

**Comment by HagegeR** [comment]
maybe try with CUDNN_HALF disabled and see if it's still happening, just to narrow the differences between the two builds

**Comment by haviduck** [comment]
i recently got  2080 ti and had the same issue, it helped setting the right arch and rebuilding. my specs are pretty close to yours.

**Comment by blaine141** [comment]
By setting arch do you mean uncommenting the arch line? I did that but did
not comment out the original arch line. Not sure if that impacts anything

**Comment by haviduck** [comment]
i commented out the original too yeah..

**Comment by blaine141** [comment]
Disabling CUDNN_half had no major impact and commenting out the first ARCH may have resulted in a slight improvement

**Comment by AlexeyAB** [comment]
@blaine141 

> That number varies but is around 0.25 on both machines. 

If `load-time` more than 0.0 than the bottleneck is in the HDD or CPU. Use SSD instead of HDD and compile Darknet with OpenCV (it will reduce CPU usage).

**Comment by blaine141** [comment]
All of my drives are SDDs and Opencv makes loading take about 25 seconds. I am currently working on reinstalling opencv

**Comment by blaine141** [comment]
Also, the Titan X machine is Ubuntu 16.04 and the 2080 ti machine is CentOS Linux release 7.7.1908

**Comment by AlexeyAB** [comment]
> and Opencv makes loading take about 25 seconds. 

Can you show screenshot?

**Comment by blaine141** [comment]
Opencv was installed via:
`sudo yum rm opencv opencv-devel opencv-python`
And then when checking version:
`pkg-config --modversion opencv
2.4.5`
And the result:
`Tensor Cores are used. (next mAP calculation at 5700 iterations) 
 5601: 6.934250, 6.934250 avg loss, 0.001000 rate, 24.000833 seconds, 358464 images
Loaded: 34.936173 seconds`
This is with OpenCV enabled

**Comment by AlexeyAB** [comment]
What is your image resolution?
What command do you use for training?
Did you use Makefile or Cmake+Make for compilation with OpenCV?
Can you show chart.png file?

**Comment by blaine141** [comment]
The images are around 5000x5000. The config has its resolution set to 1600x1600. 
The command is ./darknet detector train data/boards.data cfg/yolov3.cfg backup/yolov3_best.weights -i 1 -map -dont_show
Sometimes with the gpu specification sometimes not.
I use Makefile
I do no thave an example for a chart file from start to end but here is the beginning on one
![chart](https://user-images.githubusercontent.com/6363539/68236609-7ce3d880-ffd3-11e9-9445-85aa981b0f97.png)
The last training I did reached 93% MAP

**Comment by AlexeyAB** [comment]
> The images are around 5000x5000. The config has its resolution set to 1600x1600.

This is the reason, data augmentation is too expensive.
If your CPU has 16 logical-cores then try to change `3` to `8` there and recompile: https://github.com/AlexeyAB/darknet/blob/d628e8eab7371f136cc05ae090b904eede7f9c55/src/detector.c#L143


----

**Comment by blaine141** [comment]
What does this number mean?

**Comment by AlexeyAB** [comment]
Number of CPU-threads for data augmentation

**Comment by blaine141** [comment]
Would it be bad to turn it up to 16?

**Comment by AlexeyAB** [comment]
Just try 16 and look at the training speed.
For the faster training should be 1-2 free CPU-Cores for the main-thread.

**Comment by blaine141** [comment]
The loading time when OpenCV is enabled is now 8-16 seconds. Iteration time is the same. The time goes down again when I disable Opencv it goes to  < 0.01 Iteration time the same

**Comment by AlexeyAB** [comment]
So try to set this number higher  https://github.com/AlexeyAB/darknet/blob/d628e8eab7371f136cc05ae090b904eede7f9c55/src/detector.c#L143

Loading time should be 0, otherwise there is a bottleneck on CPU and it doesn't matter what GPU do you use.

**Comment by HagegeR** [comment]
a difficult workaround that can be to do the loading of the next batch as the GPU is working on the current one (if it's not already implemented)

**Comment by AlexeyAB** [comment]
@HagegeR it is already implemented. Loading time in the Darknet actually means `~= loading__and_augmentation_time - batch_training_time`

**Comment by blaine141** [comment]
The loading time is near 0 at around 0.00025. it is only a large time when opencv is enabled probably because I have a bad opencv. The issue is the large iteration time. It is not significantly larger on the 2080ti as expected.
